### ISSUE TITLE
Compile issue - missing java files

### ISSUE THREAD

**Issue by zbhuang** [Sixt/java-micro#5]
Has anyone successfully imported this project into IntelliJ ?
When I tried to do that, it failed complaining about gradle issue.

Second, after I manually downloaded dependent jar libraries, and compiled this project, I found a lot of java files reference this java file, com.sixt.service.framework.protobuf.RpcEnvelope.java, but it does not exist.
Same issue happens to com.sixt.service.framework.protobuf.FrameworkTest.java.

Or these two java files will be generated automatically ? Did I miss anything ?

Thanks.

**Comment by bkrahmer** [comment]
What was the issue you had with gradle when importing it?  

RpcEnvelope is generated code.  I suggest to run gradlew build from the command line.  You can also open the gradle tool window, and run Tasks | other | generateProto and generateTestProto

Sorry for the confusion.  We will add some more docs on how to properly get started soon.

**Comment by zbhuang** [comment]
Here is the error message I got.
Error:(57, 0) Cause: repository not found: C:\xxx\IntelliJProjects\java-micro-master
Thanks.

**Comment by bkrahmer** [comment]
It sounds like your intelli-j is setup wrong, or you are using it wrong.  I suggest going to File | Open | select the build.gradle in the directory of your repo, hit Ok, select 'Use gradle wrapper task configuration', and hit Ok.  That should work fine.

**Comment by zbhuang** [comment]
I was using import project last time. But this time, just did what you mentioned above, open.
I got the same error. 
By the way, Line 57 is listed below (maybe this will give you some clue ?), 
Error:(57, 0) Cause: repository not found: C:\xxx\IntelliJProjects\java-micro-master
git = org.ajoberstar.grgit.Grgit.open(file('..'))

**Comment by bkrahmer** [comment]
It's not clear to me what is causing that.  I literally just checked out the repo and built on an out-of-date intelli-j (2016.2.5) on a different computer, and had no problems (i did have to install protoc - https://github.com/google/protobuf/releases/tag/v3.2.0).  By the end of next week, we should be publishing jars so that you won't even need to check out and build the repository anyway...

**Comment by tomasulo** [comment]
It worked fine for me. This is what I did:

1. Checkout repository
2. Open IntelliJ and do File -> Open on the folder java-micro
3. Wait
4. IntelliJ will ask you which projects to import as Gradle project: Select both
5. A Gradle tool window on the right appears
7. Run Gradle task `install` to publish a jar to the local maven repository

I did not experience any compile errors.

`git = org.ajoberstar.grgit.Grgit.open(file('..'))` indicates that it could not open the Git repository. Maybe something with the checkout went wrong?

**Comment by zbhuang** [comment]
Thank bkrahmer and tomasulo for the help.
Found out the reason, my gradle is too old, 2.12.
I just found this project is upgraded to use gradle 3.3, so I went ahead to use gradle 3.3, and java-micro is imported into IntelliJ successfully, and jar files are published to the local gradle repository.
But I found still no RpcEnvelope.java file got generated.
I do NOT have protoc installed on my machine, maybe that is the reason. I will try it later.
But anyway, compilation issue is solved. Thank you all for the help.
